packages/core/src/common/workspace-utils.ts

```diff
@@ -0,0 +1,37 @@
+/********************************************************************************
+  * Copyright (C) 2018 Ericsson and others.
+  *
+  * This program and the accompanying materials are made available under the
+  * terms of the Eclipse Public License v. 2.0 which is available at
+  * http://www.eclipse.org/legal/epl-2.0.
+  *
+  * This Source Code may also be made available under the following Secondary
+  * Licenses when the conditions for such availability set forth in the Eclipse
+  * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+  * with the GNU Classpath Exception which is available at
+  * https://www.gnu.org/software/classpath/license.html.
+  *
+  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+  ********************************************************************************/
+
+import { Path } from './path';
+
+export namespace WorkspaceUtils {
+
+    /**
+     * Returns shortened workspace path if applicable
```

Comment:
It's a bit vague what "shortened workspace path" means, the function name `getShortenedPath` is also not very clear.  What about `makeTildePath` or something like that?  The doc could be:

    If `path` is a descendant of `home`, replace the home directory part of `path` with `~`.  This function is a no-op on Windows.